action_handlers/manage_bill_handler.py

```diff
@@ -241,8 +243,12 @@ def get_share_items_keyboard(bill_id, trans):
         )
         keyboard.append([refresh_btn])
         for item_id, item_name, __ in items:
+            if trans.has_bill_share(bill_id, item_id, user_id):
```

Comment:
Inconsistent state might occur if while executing loop of trans.has_bill_share(bill_id, item_id, user_id) , a share operation has updated a previously processed bill_share to now be the opposite state. E.g.

Processing first request, state of Item 1 is shared
trans.has_bill_share(bill_id, item_id, user_id) returns True
While processing Item 2, new request to unshare Item 1 occurs. Item 1 is_deleted set to True
First request returns state of Item 1 as shared when now it is unshared